### PR TITLE
Return PID from waitpid

### LINKED ISSUE #989: waitpid doesn't returned pid
Is there a reason for this? It should return a `(Pid, WaitStatus)` so you can spawn multiple processes in parallel and wait for them to finish in any order while performing specific actions on a process' completion.

### PR DESCRIPTION
**Pull request by SUPERCILEX**
Fixes #989

### CODE REVIEW

File: tests/process/wait.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ fn test_waitpid_none() {
         .expect("failed to execute child");
     unsafe { kill(child.id() as _, SIGSTOP) };
 
-    let status = process::waitpid(None, process::WaitOptions::UNTRACED)
+    let (_, status) = process::waitpid(None, process::WaitOptions::UNTRACED)
```
Comment (sunfishcode) [inline tests/process/wait.rs:21]: Could you add an assert checking that the returned pid matches `child.id()`?

**Review by sunfishcode — COMMENTED** [review]
Looks good, just one minor comment:

**Comment by sunfishcode** [comment]
Also, those CI errors look unrelated, so don't worry about those; I'll fix those separately.

File: tests/process/wait.rs (hunk 1 above)
Comment (SUPERCILEX) [reply tests/process/wait.rs]: Done.

**Comment by sunfishcode** [comment]
Thanks!

**Pull request merged**